src/transforms/reduce/mod.rs

```diff
@@ -885,4 +891,48 @@ merge_strategies.bar = "concat"
         })
         .await;
     }
+
+    #[tokio::test]
+    async fn last_flush_at() {
+        let reduce_config = toml::from_str::<ReduceConfig>(indoc! {r#"
+                group_by = ["user_id"]
+                expire_after_ms = 100
```

Comment:
Can we make this 1ms? Just to ensure that the chances of this test fails due to a race condition is low.

Reply:
This would not be the right thing to do.

In the test we set events to expire after 100ms, and we periodically submit events every 50ms.
Setting expiry to be less than the event submission period would mean that each event expire on his own, before next event shows. This does not reproduce the issue, and it means that this test will pass even without the fix in the codebase.

We test which events arrived after 220ms. Setting expiration to be greater than this value means that no event should expire at the point where we test which events arrived, thus the test will fail.
So overall we must keep`expire_after_ms` in the range `(100,220)`.
We can of course lower all values, e.g. divide them all by a factor of 10 - (5,10,22) instead of (50,100,220), however in that case code execution time might come into effect, so I made a compromise between having a stable test and having it execute in a very short  time.

As a side note, I would like to say that although this test looks as if it should be done in a second, it seems to take much longer. I haven't research that yet, and could use other's help with that.

Reply:
Alright, this looks good to me then.